python-agent-tools/incident-lookup/tool.py

```diff
@@ -0,0 +1,68 @@
+from dataiku.llm.agent_tools import BaseAgentTool
+from servicenow_client import ServiceNowClient
+from safe_logger import SafeLogger
+
+
+logger = SafeLogger("servicenow plugin", ["password"])
+
+
+class ServicenowLookupIncidentIDTool(BaseAgentTool):
+    def set_config(self, config, plugin_config):
+        self.config = config
+        self.client = ServiceNowClient(config)
+
+    def get_descriptor(self, tool):
+        descriptor = {
+            "description": "This tool can be used to search for a incident on this ServiceNow instance. The input to this tool is a dictionary containing at least one known detail about the incident to lookup, e.g. '{'email':'john.doe@example.com'}' or '{'user_name':'john.doe'}'",
+            "inputSchema": {
+                "$id": "https://dataiku.com/agents/tools/search/input",
+                "title": "Lookup Servicenow incident tool",
+                "type": "object",
+                "properties": {
+                    "description_contains": {
+                        "type": "string",
+                        "decription": "Contain search terms for the incident, for instance 'cannot login' "
+                    },
+                    "number": {
+                        "type": "string",
+                        "decription": "Contains the effective number of the incident to look up. It should start with INC folowed by sevent digits, for instance 'INC0010022'"
```

Comment:
Based on the description, the number filter should be `numberCONTAINS` as well, otherwise, we should precise

```suggestion
                        "decription": "Contains the exact effective number of the incident to look up. It should start with INC folowed by sevent digits, for instance 'INC0010022'"
```